### ISSUE TITLE
Handling specialized .weapon situations

### ISSUE THREAD

**Issue by drowzero0** [OctarineSourcerer/cyberpunk2020-foundry#53]
At the moment there is no chance to create weapons that show up with diffrent weaponmodifiers.
in this case i am talking about RAMJET Weapons.
![image](https://user-images.githubusercontent.com/62395017/114915861-6f678a80-9e24-11eb-8853-b8d7be19f9ba.png)

to make it possible to handle every distanceclass with correct damage i made this workaround for now:
![image](https://user-images.githubusercontent.com/62395017/114915964-8c9c5900-9e24-11eb-9ed6-4bf685a3d384.png)

if there is a chance to get this handled by the weaponsheet or the roll.weaponattack popup would be nice.

just something i want to mention for the future.

**Comment by OctarineSourcerer** [comment]
Wow that really is a fun case, and a neat workaround. I'd say it was an edge case, buuuut shotgun shells do something similar, albeit less effective further out. I intend to sort something out on that front, though it might not be particularly soon, as I've gotta figure out how to do it without cluttering up the UI much.

**Comment by drowzero0** [comment]
as far as i can remember there is just this one and a 2nd one the Reimei as Ramjet guns.
but i can live with my workaround for a while. just creating those is a bit painfull :D
glad there is copy paste.
but you are right, the damagefalloff from shotguns creates a nearly similar case.
but there werent so many diffrent ranges if i remember correctly?

btw i like how it looks in the charactersheet, the player knows exactly which weapon to press:
![image](https://user-images.githubusercontent.com/62395017/114919966-367de480-9e29-11eb-8872-8a5a7ec2856e.png)

(iknow there is no possibility for ET-RamJet, but this was faster than the discussion)

**Comment by OctarineSourcerer** [comment]
Oh wow! I think I might end up having a dropdown/expandable section in the weapon sheet settings like "specialist upgrade" or something like that, where you can check/enable things like ET or different damage per range. It being collapsed means most people aren't overwhelmed with it unless they go looking.

**Comment by OctarineSourcerer** [comment]
> Not sure if this applies in specialized weapon cases, as mentioned above shotgun rules are a beast of their own, but using the "autoshotgun" option in the drop down menu doesn't actually allow you to fire full auto or burst. I think technically they're supposed to be used as area denial instead of a traditional weapon, but the option would still be needed for slugs and other shell types.
> 
> Also in the weapon drop down the flamethrower option is listed as "flamethrow". Not sure if that's intentional so you can use it for fire based grenades or general fire damage but as is the wording seems odd.

These are all good points! I'll get on them :)

**Comment by drowzero0** [comment]
i just saw in the dev-branch:
![image](https://user-images.githubusercontent.com/62395017/119231987-b29fd200-bb23-11eb-88ab-ca20608f7c07.png)
this fixes this problem!
guess that fixes this mentioned problem.

**Comment by OctarineSourcerer** [comment]
Not quite! That's currently not actually taken into account in any rolls, and shouldn't be saying "DamageShort" :P but that's the start of it, yes

That, and I need to figure out some way to _not_ have expandables like that collapse after one edit.

**Comment by Emilio1922** [comment]
I'm not entirely sure if this should be it's own issue, but I can see it helping here too.
Some weapons, like those with folding stocks have different WA values depending on whether or not the stock is folded. Also some weapons have accuracy modifiers based on whether they're firing in single fire or three round burst(albeit stuff like this is rare.)
What could be helpful is an "advanced" weapon item for fringe cases where you can set up multiple weapon settings as profiles, then have them as a drop down on the attack card so a user can select which one he's using. This would mean naming the setting types would be important and I don't know how complicated all this would be to implement vs just add it to the extra modifiers when attacking or using separate items, but it would allow for modifying WA, damage, range, etc. based on weapon mods being in use or not, weapons like the ramjet, or weapons like the multi ammo revolver in Protect & Serve, where you can select and switch ammo types on the fly, all with different characteristics.

**Comment by OctarineSourcerer** [comment]
> I'm not entirely sure if this should be it's own issue, but I can see it helping here too.
> 
> Some weapons, like those with folding stocks have different WA values depending on whether or not the stock is folded. Also some weapons have accuracy modifiers based on whether they're firing in single fire or three round burst.
> 
> What could be helpful is an "advanced" weapon item for fringe cases where you can set up multiple weapon settings, then have them as a drop down or checkbox on the attack card so a user can select which one he's using. This would mean naming the setting types would be important and I don't know how complicated all this would be to implement vs just add it to the extra modifiers when attacking or using separate items, but it would allow for modifying WA, damage, range, etc. based on weapon mods being in use or not, weapons like the ramjet, or weapons like the multi ammo revolver in Protect & Serve, where you can select and switch ammo types on the fly, all with different characteristics.

Am I right in interpreting this as essentially providing alternatives for each of the stats the weapon has that they can select? eg alternate damages, alternate range, WA etc?

**Comment by Emilio1922** [comment]
Sorry that comment should have been reedited before posting rather than during.

It's essentially making duplicate weapon settings tabs in the gear menu so the person making the item in the compendium can make two completely separate settings profiles for the same weapon, but making it easy on the players in play because they just pick what they're using in a drop down. 

If you'll forgive the very quick mockup in GIMP, an FN RAL with an attached grenade launcher might look like this,
![Garbo](https://user-images.githubusercontent.com/32806614/119699193-e3338480-be1f-11eb-9d72-161f06ddbe91.png)
with each tab having a fully editable settings menu. Then you could pick which you were using when attacking. In this case the weapons would have two totally different sets of stats.

But in this one I'd use the same stats, only changing the WA and Concealability stat.
![Garbo2](https://user-images.githubusercontent.com/32806614/119699885-9e5c1d80-be20-11eb-9012-39ef91624539.png)

**Comment by OctarineSourcerer** [comment]
Oh okay, that's pretty cool! Hmm. I might do something close - a selector of weapon profiles within one tab (it'd be the easiest to do under the hood, and would preserve the current meaning of the tabs). The slightly trickier thing would be adding weapon profiles, but I reckon that's doable.

I don't know if they cover all of this - I'd rather the user not have to select the correct profile when they fire, so there may still be some more hardcoded options. But it's a very very good option and absolutely hits the case of weapons with multiple user-choosable fire modes.

EDIT: okay after thinking about this, I've got an important question: in what way does this work better than just having multiple items? It's essentially a duplicate but changed set of options, which is similar to duplicating and changing an item; but with more complication under the hood, and a change of where the user would select things.

**Comment by Emilio1922** [comment]
In response to why do this instead of just having multiple items, for the same reason as having damage at ranges for the ramjet and a handful of shotguns, it makes the inventory screen a bit less cluttered than having "Rifle, stock folded. Rifle, stock unfolded. Rifle, stock unfolded full auto," all as separate items with just a change to WA or range. It also means you don't need to worry about weight only being accounted for on one version of the gun. Basically it solves the issue on the two niche weapons types that the advanced options you added solved, plus giving a lot of options to the GM/players on existing weapons and stuff they homebrew.

Honestly even if it was done in a way where it puts multiple items in the combat tab for all the variations but keeps the gear tab clean with only one item per weapon I think it would be a nice improvement. The gear tab can look a bit cluttered if you're filling it with a dozen versions of a weapon with a folding stock and different full auto WA and damages at ranges and all the non combat gear a PC might need, whereas the combat tab keeps relatively clean thanks only including the weapons as items and being set in one column.

That said I have no idea how this would look on the back end, it's very much a "What I would like the front end to look like" kind of deal mocked up in GIMP in 10 seconds. Like I said in the first post, I have no experience programming and if this is a lot more work than it's worth it might not be better than just having a list of items in the combat and gear tab.

**Comment by OctarineSourcerer** [comment]
No, that's a totally reasonable point, and it gets even more annoying when thinking about deleting items.
I'd say multiple weapon profiles is an okay way of going about it, with a default one being selectable. I'm happy to try doing it.

It might get fairly complicated under the hood depending on how I go about it, and may require some design thought - this will be a little while in coming, and definitely won't be in the next release. But I intend on doing it :)
